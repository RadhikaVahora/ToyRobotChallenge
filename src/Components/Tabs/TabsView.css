

.Rectangle1 {
  position: absolute;
  width: 620px;
  height: 249px;
  left: 201px;
  top: 62px;
  
  background: #FFFFFF;
  border: 1px solid #E7E7E8;
  box-sizing: border-box;

}


.TabLayout{
  display: flex;
  list-style-type: none;
  flex-direction: row;
  justify-content: space-around; 
  width: 580px;
  height: 20px;
}

