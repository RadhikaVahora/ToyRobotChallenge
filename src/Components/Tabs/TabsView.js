import React from 'react';
import TabView from '../Tab/TabView';
import './TabsView.css';
import Config from '../../Constants/Const';

class TabsView extends React.PureComponent {
    state = {
        activeTab:'Tincidunt',
    }

    onClickTabItem = (tab) =>{
        this.setState({
            activeTab:tab
        });
    }

    render(){
        return(
            <div>
                <h1 className='Title'>Tab Component</h1>
                <div className="Rectangle1">
                <ul className='TabLayout'>
                    {Config.TABNAME.map((element,index) =>(
                        <TabView 
                        key={index}
                        label ={element.name}
                        activeTab={this.state.activeTab}
                        onTabClick = {this.onClickTabItem}/>

                    ))}
                </ul>
                </div>
            </div>
        );
    }
}
export default TabsView;