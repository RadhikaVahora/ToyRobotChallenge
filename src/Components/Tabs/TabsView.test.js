
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TabsView from './TabsView';

describe('TabsView', () => {
    it('should render all tabs correctly',() =>{
        const renderer = new ShallowRenderer();
        const wrapper = renderer.render(<TabsView />);
        expect(wrapper).toMatchSnapshot();
    });
});