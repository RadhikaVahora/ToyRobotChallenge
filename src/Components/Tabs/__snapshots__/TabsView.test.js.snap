// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsView should render all tabs correctly 1`] = `
<div>
  <h1>
    Tab Component
  </h1>
  <div
    className="Rectangle1"
  >
    <ul
      className="TabLayout"
    >
      <TabView
        activeTab="Tincidunt"
        label="Tincidunt"
        onTabClick={[Function]}
      />
      <TabView
        activeTab="Tincidunt"
        label="Sodales"
        onTabClick={[Function]}
      />
    </ul>
  </div>
</div>
`;
